container/image.bzl

```diff
@@ -244,35 +194,41 @@ def _impl(ctx, files=None, file_map=None, empty_files=None, directory=None,
   cmd = cmd or ctx.attr.cmd
   symlinks = symlinks or ctx.attr.symlinks
   output = output or ctx.outputs.executable
-  env = env or ctx.attr.env
+  layers = container_layers or ctx.attr.container_layers
   debs = debs or ctx.files.debs
   tars = tars or ctx.files.tars
 
   # Generate the unzipped filesystem layer, and its sha256 (aka diff_id).
-  unzipped_layer, diff_id = _build_layer(ctx, files=files, file_map=file_map,
-                                         empty_files=empty_files,
-                                         directory=directory, symlinks=symlinks,
-                                         debs=debs, tars=tars)
+  unzipped_layer, diff_id = build_layer(ctx, files=files, file_map=file_map,
+                                        empty_files=empty_files,
+                                        directory=directory, symlinks=symlinks,
+                                        debs=debs, tars=tars)
 
   # Generate the zipped filesystem layer, and its sha256 (aka blob sum)
-  zipped_layer, blob_sum = _zip_layer(ctx, unzipped_layer)
+  zipped_layer, blob_sum = zip_layer(ctx, unzipped_layer)
+
+  # Get the layers and shas from our base.
+  # These are ordered as they'd appear in the v2.2 config,
+  # so they grow at the end.
+  parent_parts = _get_layers(ctx, ctx.attr.base, ctx.files.base)
+  zipped_layers = parent_parts.get("zipped_layer", []) + _getLayerInfo(layers, "zipped_layer") + [zipped_layer]
+  shas = parent_parts.get("blobsum", [])  + _getLayerInfo(layers, "blob_sum") + [blob_sum]
+  unzipped_layers = parent_parts.get("unzipped_layer", []) + _getLayerInfo(layers, "unzipped_layer") + [unzipped_layer]
+  layer_diff_ids = _getLayerInfo(layers, "diff_id") + [diff_id]
+  diff_ids = parent_parts.get("diff_id", []) + layer_diff_ids
+
+  # Get and merge environment variables
+  env = {}
+  [env.update(layer[LayerInfo].env) for layer in layers]
```

Comment:
This doesn't work, it doesn't properly evaluate read-modify-write environment variables like `PATH`.

I believe that for this to work properly, we actually want N calls to `_image_config(...)`.

Reply:
It looks like _image_config will deal with more things than envvars. 

I wrote a skylark function `_evalute_merge_env ` to handle envvar here properly. And I have unit test to cover the read-modify-write case like `PATH`.

Reply:
> It looks like _image_config will deal with more things than envvars.

That's fine, so just pass env vars :)

> I wrote a skylark function _evalute_merge_env to handle envvar here properly

I'd rather not fork this logic across two places.  I'd definitely prefer we use `_image_config` iteratively.